connectors/sync_job_runner.py

```diff
@@ -189,11 +189,11 @@ async def execute(self):
             while not self.sync_orchestrator.done():
                 await self.check_job()
                 await asyncio.sleep(JOB_CHECK_INTERVAL)
-            fetch_error = self.sync_orchestrator.fetch_error()
-            sync_status = (
-                JobStatus.COMPLETED if fetch_error is None else JobStatus.ERROR
-            )
-            await self._sync_done(sync_status=sync_status, sync_error=fetch_error)
+                if self.sync_orchestrator.get_error() is not None:
+                    break
```

Comment:
IMO, if the sync_orchestrator hits an error, we should terminate the job - not keep running until "done".

Reply:
`done` here actually repeats "done" concept of task object: https://docs.python.org/3/library/asyncio-task.html#asyncio.Task.done

> A Task is done when the wrapped coroutine either returned a value, raised an exception, or the Task was cancelled.

So this code will theoretically never execute.

Reply:
hmm. I'll have to try again, but I'm pretty sure that I'd put this here because I was seeing ERROR logs every 500 docs, but the sync didn't stop until the Extractor stopped extracting (for my slack test, after fetching 30k docs, but literally 0 of them being able to get to ES)

Reply:
Ok, I see how this is _supposed_ to repeat the async "done" concept, but it has a logic bug currently:

```
    def done(self):
        if self._extractor_task is not None and not self._extractor_task.done():
            return False
        if self._sink_task is not None and not self._sink_task.done():
            return False
        return True
```

This short-circuits and returns `false` if just the first condition is met. So if the Sink has an error, we still say "we're not done!" because the  the Extractor isn't done.

So I'll just move the `get_error()` check _here_, so that the "done" pattern can still be followed, but we avoid the logical issue. Sound good?